Data Prep Code.Rmd

```diff
@@ -0,0 +1,317 @@
+kruskal_BDG <- kruskal.test(kept ~ moc, data = BDG)
+print("Kruskal-Wallis test for Bully Dictator Game:")
+print(kruskal_BDG)
+
+# Perform Mann-Whitney U test for BDG choice
+group_C <- BDG[BDG$treatment_group == "C", "kept"]$kept
+group_TD <- BDG[BDG$treatment_group == "TD", "kept"]$kept
+group_MD <- BDG[BDG$treatment_group == "MD", "kept"]$kept
+
+# Perform Mann-Whitney U test between TD and MD for BDG
+# This test compares the distribution of 'kept' between TD and MD within the Bully Dictator Game
+result_TD_MD <- wilcox.test(group_TD, group_MD)
+z_value_TD_MD <- qnorm(1 - result_TD_MD$p.value / 2)  # Convert p-value to z-score
+p_value_TD_MD <- result_TD_MD$p.value
+cat(sprintf("Mann-Whitney U test for BDG between TD and MD, z = %.3f, p = %.4f.\n", z_value_TD_MD, p_value_TD_MD))
+
+# Perform Mann-Whitney U test between C and MD for BDG
+# This test compares the distribution of 'kept' between C and MD within the Bully Dictator Game
+result_C_MD <- wilcox.test(group_C, group_MD)
+z_value_C_MD <- qnorm(1 - result_C_MD$p.value / 2)
+p_value_C_MD <- result_C_MD$p.value
+cat(sprintf("Mann-Whitney U test for BDG between C and MD, z = %.3f, p = %.4f.\n", z_value_C_MD, p_value_C_MD))
+
+# Perform Mann-Whitney U test between C and TD for BDG
+# This test compares the distribution of 'kept' between C and TD within the Bully Dictator Game
+result_C_TD <- wilcox.test(group_TD, group_C)
+z_value_C_TD <- qnorm(1 - result_C_TD$p.value / 2)
+p_value_C_TD <- result_C_TD$p.value
+cat(sprintf("Mann-Whitney U test for BDG between C and TD, z = %.3f, p = %.4f.\n", z_value_C_TD, p_value_C_TD))
+
+```
+
+```{r Figure 4, echo=FALSE}
+# Prepare data with renaming and calculating statistics
+dictator_game_data <- df %>%
+  mutate(game_type = case_when(
+    game_type == "Standard Dictator Game" ~ "DG",
+    game_type == "Bully Dictator Game" ~ "BDG"
+  )) %>%
+  mutate(treatment_group = case_when(
+    moc == 0 ~ "C",
+    moc == 1 ~ "TD",
+    moc == 2 ~ "MD"
+  )) %>%
+  group_by(game_type, treatment_group) %>%
+  summarise(
+    Mean_Got = round(mean(kept, na.rm = TRUE), 2),
+    SD = round(sd(kept, na.rm = TRUE), 2),
+    SE = round(sd(kept, na.rm = TRUE) / sqrt(n()), 2) # Standard error
+  ) %>%
+  ungroup() %>%
+  mutate(game_type = factor(game_type, levels = c("DG", "BDG"))) %>%
+  arrange(game_type, desc(Mean_Got))
+
+# Print the adjusted data
+print(dictator_game_data)
+
+# Plot the data
+ggplot(dictator_game_data, aes(x = treatment_group, y = Mean_Got, fill = game_type)) +
+  geom_bar(stat = "identity", position = position_dodge(), width = 0.7, alpha=0.9) +
+  geom_errorbar(aes(ymin = Mean_Got - SE, ymax = Mean_Got + SE),
+                position = position_dodge(0.7), width = 0.25) +
+  scale_fill_manual(values = c("DG" = "yellow", "BDG" = "grey")) +
+  labs(title = "Figure 4: Mean of Got by Treatment Group and Game Type",
+       y = "Mean of Got") +
+  theme_minimal() +
+  theme(plot.title = element_text(hjust = 0.5))

```

Comment:
Please use separate chunks for printing text/data/plots and write a comments as a plain text between those chunks.